### PR TITLE
Read from CB API and create AEC

### PR DESCRIPTION
**Pull request by nanditagoclimate**
- [X] Create company input(CI)
- [X] Read from CB API
- [X] Convert accounting values to emissions
- [X] Overwrite CB values with CI values
- [X] Create AEC
- [X] Test cases(Woohoo!)
- [X] Handle duplicates
- [X] Test code with static data

As a part of different PR:
Handle monthly emissions

### CODE REVIEW

**Review by kallenilver — APPROVED** [review]
Nice work! Relatively easy to understand 👍 

Would be nice with some tests structure too, especially on the calculations and especially now that you are leaving.

File: app/models/climate_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    accounting_datas.each do |accounting_data|
+      @start_date = accounting_data['start_date']
+      @end_date = accounting_data['end_date']
+      emission_sources = accounting_data['emission_sources']
+      emission_sources.each do |emission_source|
+        next unless emission_source['category'].present? && emission_source['spend'].present?
+
+        mapped_emission_factor = category_to_emission_factor_mapping[emission_source['category']]
+        if emission_factors[mapped_emission_factor] == nil
+          @coming_as_nil << emission_source['category']
```
[the remark this answers is not in the document]
Comment (nanditagoclimate) [reply app/models/climate_calculator.rb:27]: Absolutely. This was actually for me to test. We should have a better way to handle this. 
The API returned some labels that was not mapped into categories. When testing with Berge we found ["taxes_and_insurance", "cloud_services_green", "lpg"] to not have been mapped into any categories so their emissions was getting summed up and set to 'nil'

File: app/models/climate_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        end
+        next unless emission_factors[mapped_emission_factor]
```
[the remark this answers is not in the document]
Comment (nanditagoclimate) [reply app/models/climate_calculator.rb:29]: Agreed.

**Comment by nanditagoclimate** [comment]
Will make the changes as per the comments, clean up the code and add some tests.

**Comment by nanditagoclimate** [comment]
Test cases are failing cos this code only looks at locally hosted API. I will work on integrating with the API via DNS next and then make changes in the code and merge them. cc: @kallenilver

File: app/controllers/admin/company_inputs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+module Admin
+  class CompanyInputsController < AdminController
+    before_action :set_company_input, only: [:show, :update, :edit, :destroy]
+
+    def index
+      @company_inputs = CompanyInput.all.order(created_at: :asc)
+    end
+
+    def show
+      @company_input = CompanyInput.find(params[:id])
```
Comment (kallenilver) [inline app/controllers/admin/company_inputs_controller.rb:12]: I guess this is redundant if it's done already in before_action?

**Review by kallenilver — APPROVED** [review]
Good job! 🥳 
I don't have the mind space to go through all the calculations, but very happy to see tests in place 👍

**Pull request merged**